datasets/plugins/batch/batch_flow_dataset_plugin.py

```diff
@@ -0,0 +1,57 @@
+from typing import TYPE_CHECKING, Iterable, Optional, Tuple, Union
+
+from datasets.context import Context
+from datasets.dataset_plugin import DatasetPlugin
+from datasets.plugins import BatchDatasetPlugin
+
+
+if TYPE_CHECKING:
+    from metaflow import Run
+
+
+@DatasetPlugin.register_plugin(constructor_keys={"flow_dataset"}, context=Context.BATCH)
+class BatchFlowDatasetPlugin(BatchDatasetPlugin):
+    def __init__(
+        self,
+        flow_dataset: str,
+        name: str = None,
+        columns: Optional[Union[Iterable[str], str]] = None,
+        run_id: Optional[str] = "latest_successful_run",
+        class_field_name: Optional[str] = None,
+    ):
+
+        self.flow_dataset = flow_dataset
+        self.flow_name, self.dataset_name = flow_dataset.split(".")
+
+        run, run_id = _get_run_id(self.flow_name, run_id)
+
+        dataset = getattr(run.data, self.dataset_name)
+
+        super(BatchFlowDatasetPlugin, self).__init__(
+            name=dataset.name,
+            logical_key=dataset.key,
+            columns=columns,
+            run_id=run_id,
+            class_field_name=class_field_name if class_field_name else (name if name else self.dataset_name),
+        )
+        # The program name is that of the original dataset name
+        self.program_name = dataset.program_name
+
+
+def _get_run_id(flow_name: str, run_id: Optional[str]) -> Tuple["Run", Optional[str]]:
+    from metaflow import Flow, Run
+
+    flow = Flow(flow_name)
+    run: Run
+    ret_run_id: Optional[str]
+    if run_id in ["latest_successful_run", "latest_run"]:
+        run = flow.latest_run if run_id == "latest_run" else flow.latest_successful_run
```

Comment:
`run = getattr(flow, run_id)`?

Reply:
This `flow ` is the representation of the flow, from https://docs.metaflow.org/metaflow/client

Reply:
Going to resolve as either way works, I just thought that approach was a little nicer but definitely opinionated so can go either way 😄

Reply:
oh `run = getattr(flow, run_id)` wouldn't work... you have to first get the `Run`...

Reply:
Oh I meant literally as a direct replacement for line 48 😅  You can just shorten this:

`run = flow.latest_run if run_id == "latest_run" else flow.latest_successful_run`

to:

`run = getattr(flow, run_id)`

And it'd be functionally the same. Doesn't really matter too much though and there is some argument that the statement above is easier to read!

Reply:
ah I see how I got confused!  And I'd have to rename the variable to `run = getattr(flow, latest_run_type)` something

Reply:
Yeah I guess that's the risk taken with overloading the `run_id` to mean more than just that specific concept but from my understanding that is a Metaflow core decision 🤷